components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/processor/JWTInboundRequestProcessor.java

```diff
@@ -0,0 +1,626 @@
+import org.wso2.carbon.identity.application.mgt.ApplicationManagementService;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.sso.jwt.exception.JWTIdentityException;
+import org.wso2.carbon.identity.sso.jwt.message.JWTInboundRequest;
+import org.wso2.carbon.identity.sso.jwt.message.JWTInboundResponse;
+import org.wso2.carbon.identity.sso.jwt.util.JWTInboundConstants;
+import org.wso2.carbon.identity.sso.jwt.util.JWTInboundUtil;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The processor class represents the core functionality od the inbound authenticator. Being a subclass of the
+ * IdentityProcessor class, the developer is required to override certain methods which dictate the functional
+ * elements.
+ *
+ * In the process method, the developer is able to leverage standard methods offered by the framework to send the
+ * request on to the framework (buildResponseForFrameworkLogin()) as well as handle the response after going through
+ * the authentication step (processResponseFromFrameworkLogin()).
+ *
+ * The getRelyingPartyId() method is used for correlating the protocol of the incoming authentication request
+ * (i.e. the protocol represented by this processor) against a particular SP in the Identity Server which contains the
+ * information related to the actual authentication, either through local and/or federated authenticators.
+ *
+ * This class is extensible either by implementing the IdentityProcessor or extending the
+ * JWTInboundRequestProcessor. The customized class should be defined under
+ * <JWTSSO><RequestProcessor></RequestProcessor></JWTSSO> in identity.xml. Otherwise JWTInboundRequestProcessor will be
+ * used as the default request processor.
+ */
+public class JWTInboundRequestProcessor extends IdentityProcessor {
+
+    private static Log log = LogFactory.getLog(JWTInboundRequestProcessor.class);
+
+    private AbstractInboundAuthenticatorConfig jwtInboundAuthConfig = null;
+
+    private String relyingParty;
+    private String redirectUrl;
+    private String errorUrl;
+    private String endpointUrl;
+    private JWSAlgorithm jwsAlgorithm;
+
+    public JWTInboundRequestProcessor(AbstractInboundAuthenticatorConfig jwtInboundAuthConfig) {
+
+        this.jwtInboundAuthConfig = jwtInboundAuthConfig;
+    }
+
+    @Override
+    public String getName() {
+
+        return jwtInboundAuthConfig.getName();
+    }
+
+    @Override
+    public String getCallbackPath(IdentityMessageContext identityMessageContext) {
+
+        String endpoint = "identity";
+        String tenantDomain = identityMessageContext.getRequest().getTenantDomain();
+        if (!tenantDomain.equals("carbon.super")) {
```

Comment:
Check if super tenant domain can be taken from a common constant
one possibility if available `MultitenantConstants.SUPER_TENANT_DOMAIN_NAME`

Reply:
Fixed